File: cases/taint/go/infeasible-branch-negative/infeasible_branch_negative.go
package dataflowbench

func dfb_source() int { // DFB-SOURCE: infeasible-branch-input
	return 1
}

func dfb_sink(value int) {} // DFB-SINK: infeasible-branch-sink

func run() {
	value := 0
	if false {
		value = dfb_source() // DFB-WITNESS: infeasible-tainted-branch
	}
	dfb_sink(value)
}

File: cases/taint/scala/infeasible-branch-negative/InfeasibleBranchNegative.scala
package dataflowbench

object InfeasibleBranchNegative {
  def dfb_source(): Int = { // DFB-SOURCE: infeasible-branch-input
    1
  }

  def dfb_sink(value: Int): Unit = {} // DFB-SINK: infeasible-branch-sink

  def run(): Unit = {
    var value = 0
    if (false) {
      value = dfb_source() // DFB-WITNESS: infeasible-tainted-branch
    }
    dfb_sink(value)
  }
}
